### ISSUE TITLE
Rule Change: `accessor-pairs` and `grouped-accessor-pairs` should support types

### ISSUE THREAD

**Issue by DMartens** [eslint/eslint#19860]
### What rule do you want to change?

accessor-pairs and grouped-accessor-pairs

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// accessor-pairs
interface I {
  get prop(): any
}

// grouped-accessor-pairs
interface I {
  get prop(): any
  propInBetween: true
  set prop(value: any): any
}
```

### What does the rule currently do for this code?

Nothing as only object properties and class members are linted.

### What will the rule do after it's changed?

Also check accessors in types (`TSTypeLiteral` and `TSInterfaceDeclaration`) if enabled by an option (similar to `enforceForClassMembers`).

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

As more core rules are now TypeScript-aware, I hope these changes are welcome as `typescript-eslint` does not offer extension rules for the accessor rules.

**Comment by Tanujkanti4441** [comment]
Hi @DMartens, thanks for the suggestion.

I am in favor of adding this. I think it's good to add options to enable checking `accessor-pairs` inside both `TSTypeLiteral` and `TSInterfaceDeclaration`. We can name them - `enforceForTSTypeLiteral` and `enforceForTSInterfaceDeclaration`.

Let's have some more opinions on this from @eslint/eslint-team.

**Comment by nzakas** [comment]
I'm in favor. :+1:

Some questions:

- Should these checks be on by default?
- Does it makes sense to individually configure this for types and interfaces? Or would we just have a `enforceForTSTypes` option that covers both?

**Comment by DMartens** [comment]
I would disable the additional checks by default because more errors are a breaking change. Then enable them by default in the next major eslint version.
I also think two options would be overkill.

**Comment by JoshuaKGoldberg** [comment]
👍 all around from me on what's been said: 

* Disabling by default in this version, then enabling by default in the next version
* Keeping to just one option

For the option name, maybe `enforceForTypes`? I think it's good to not be TypeScript-specific, in the hope that eventually some other typed language comes along.

**Comment by nzakas** [comment]
> For the option name, maybe `enforceForTypes`? I think it's good to not be TypeScript-specific, in the hope that eventually some other typed language comes along.

I'm of the opposite opinion. I think *especially* if other typed languages come along, we need to be clear that this only works for TypeScript types. If those other languages produce a parser that generates `TS*` nodes, then that will be the indication that it's compatible with TypeScript types.

**Comment by nzakas** [comment]
Marking as accepted.
